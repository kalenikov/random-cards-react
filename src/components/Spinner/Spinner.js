import CircularProgress from "@material-ui/core/CircularProgress"
import LinearProgress from "@material-ui/core/LinearProgress"
import React from 'react'

const Spinner = () => {
    // return <CircularProgress />
    return <LinearProgress />
}

export default Spinner
