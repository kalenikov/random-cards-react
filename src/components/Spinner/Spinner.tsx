import CircularProgress from "@material-ui/core/CircularProgress"
import Container from '@material-ui/core/Container'
import LinearProgress from "@material-ui/core/LinearProgress"
import {makeStyles} from "@material-ui/core/styles"
import React from 'react'
import {Typography} from "@material-ui/core"

const useStyles = makeStyles(theme => ({
    root: {
        height: '97vh',
        // backgroundColor: '#a664ff',
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'center',
        justifyContent: 'center'
    },
    spinner: {
        marginBottom: '30px'
    }
}))

const Spinner = ({title = 'Songs is loading...'}) => {
    const styles = useStyles()

    return (
        <Container className={styles.root}>
            <CircularProgress size={100} className={styles.spinner} />
            <Typography variant={'h6'}>{title}</Typography>
        </Container>
    )
}

export default Spinner
