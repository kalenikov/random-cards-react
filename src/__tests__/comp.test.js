import React from 'react'
import Spinner from '../components/Spinner/Spinner'

describe('test', ()=>{
    it('test 2', ()=>{
        const wrapper = render(<Spinner title="unique1" />);
        expect(wrapper).toMatchSnapshot()
    })
})
