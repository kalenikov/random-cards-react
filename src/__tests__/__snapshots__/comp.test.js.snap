// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test test 2 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "MuiContainer-root makeStyles-root-1 MuiContainer-maxWidthLg",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "MuiCircularProgress-root makeStyles-spinner-2 MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate",
          "role": "progressbar",
          "style": "width:100px;height:100px",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "MuiCircularProgress-svg",
              "viewBox": "22 22 44 44",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate",
                  "cx": "44",
                  "cy": "44",
                  "fill": "none",
                  "r": "20.2",
                  "stroke-width": "3.6",
                },
                "children": Array [],
                "name": "circle",
                "namespace": "http://www.w3.org/2000/svg",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "cx": undefined,
                  "cy": undefined,
                  "fill": undefined,
                  "r": undefined,
                  "stroke-width": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "cx": undefined,
                  "cy": undefined,
                  "fill": undefined,
                  "r": undefined,
                  "stroke-width": undefined,
                },
              },
            ],
            "name": "svg",
            "namespace": "http://www.w3.org/2000/svg",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "viewBox": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "viewBox": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "MuiTypography-root MuiTypography-h6",
          },
          "children": Array [
            Object {
              "data": "unique1",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "h6",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "role": undefined,
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "role": undefined,
          "style": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "MuiTypography-root MuiTypography-h6",
        },
        "children": Array [
          Object {
            "data": "unique1",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "h6",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "MuiCircularProgress-root makeStyles-spinner-2 MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate",
            "role": "progressbar",
            "style": "width:100px;height:100px",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "MuiCircularProgress-svg",
                "viewBox": "22 22 44 44",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate",
                    "cx": "44",
                    "cy": "44",
                    "fill": "none",
                    "r": "20.2",
                    "stroke-width": "3.6",
                  },
                  "children": Array [],
                  "name": "circle",
                  "namespace": "http://www.w3.org/2000/svg",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "cx": undefined,
                    "cy": undefined,
                    "fill": undefined,
                    "r": undefined,
                    "stroke-width": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "cx": undefined,
                    "cy": undefined,
                    "fill": undefined,
                    "r": undefined,
                    "stroke-width": undefined,
                  },
                },
              ],
              "name": "svg",
              "namespace": "http://www.w3.org/2000/svg",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "viewBox": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "viewBox": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "role": undefined,
            "style": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "role": undefined,
            "style": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
